### ISSUE TITLE
Problems Generating Call Graph for Apps

### ISSUE THREAD

**Issue by ShengliangD** [arguslab/Argus-SAF#58]
I'm start using Argus-SAF to generate call graphs. In my case, both callbacks and ICC communications are supposed to be viewed as edges in call graphs. I'm reading about the tutorial and examples to learn about it. But I'm wondering if there is some simple solution or direct example of achieving this.

Actually, I found that there are some mismatches between the homepage and the latest version of Argus-SAF, e.g., there is no such field `resolve_icc`. To my current understanding, I can generate call-graph with `InterProceduralSuperSpark` for each component, and then merge them together. But as far as I can see from the result, there is no edge corresponding to ICC communications in the merged call graph(using `InterProceduralDataflowGraph.merge()`). Also, there are no edges for callbacks like `Thread.run()` or `Thread.start()`.

To summarize, I'm wondering:
1. if there is direct way to generate call graphs with callbacks and ICCs considered;
2. if this must be done by hand, it would be nice to have some simple examples.

I'd be very grateful if someone can help.

**Comment by ShengliangD** [comment]
I read the paper and started to getting the points. Hope I can solve it.
